chained_swapper/tests/test_chained_swapper.py

```diff
@@ -0,0 +1,184 @@
+                "field_id": cls.env.ref(
+                    "product.field_product_product__product_tmpl_id"
+                ).id,
+            }
+        )
+        cls.chained_swapper_text = cls.env["chained.swapper"].create(
+            {
+                "name": "Product",
+                "model_id": cls.env.ref("product.model_product_product").id,
+                "field_id": cls.env.ref(
+                    "product.field_product_product__description"
+                ).id,
+            }
+        )
+
+    def test_create_unlink_action(self):
+        """Test if Sidebar Action is added / removed to / from given object."""
+        action = (
+            self.chained_swapper.ref_ir_act_window_id
+            and self.chained_swapper.ref_ir_act_window_id.binding_model_id
+        )
+        self.assertTrue(action)
+        # Remove the action
+        self.chained_swapper.unlink_action()
+        action = self.chained_swapper.ref_ir_act_window_id
+        self.assertFalse(action)
+        # Add an action
+        self.chained_swapper.add_action()
+        action = (
+            self.chained_swapper.ref_ir_act_window_id
+            and self.chained_swapper.ref_ir_act_window_id.binding_model_id
+        )
+        self.assertTrue(action)
+
+    def test_unlink_chained_swapper(self):
+        """Test if related actions are removed when a chained swapper
+        record is unlinked."""
+        action_id = self.chained_swapper.ref_ir_act_window_id.id
+        self.chained_swapper.unlink()
+        action = self.env["ir.actions.act_window"].search([("id", "=", action_id)])
+        self.assertFalse(action)
+
+    def test_change_constrained_partner_language(self):
+        with self.assertRaises(exceptions.UserError):
+            self.env["chained.swapper.wizard"].with_context(
+                active_model="res.partner",
+                active_id=self.partner_parent.id,
+                active_ids=(self.partner_parent | self.partner_child_1).ids,
+                chained_swapper_id=self.chained_swapper.id,
+            ).create({"lang": "es_ES"})
+
+    def test_change_partner_language(self):
+        self.env["chained.swapper.wizard"].with_context(
+            active_model="res.partner",
+            active_id=self.partner_parent.id,
+            active_ids=[self.partner_parent.id],
+            chained_swapper_id=self.chained_swapper.id,
+        ).create({"lang": "es_ES"})
+        self.assertEqual(self.partner_parent.lang, "es_ES")
+        self.assertEqual(self.partner_child_1.lang, "es_ES")
+        self.assertEqual(self.partner_child_2.lang, "es_ES")
+        chained_many2one = (
+            self.env["chained.swapper.wizard"]
+            .with_context(
+                active_model="product.product",
+                chained_swapper_id=self.chained_swapper_inherits.id,
+            )
+            .create({"product_tmpl_id": 1})
+        )
+        chained_many2one.fields_view_get()
```

Comment:
There's still call to this method + code that doesn't test the real purpose of the module.

Reply:
In this test case, I should not use fields_view_get or use it in separate testing?

Reply:
You are not adding anything calling it, as this is an internal method called by action controller in the web client, so you are not testing anything real even if you call it manually and test the return result.

Reply:
So should I consider fields_view_get method in test cases or not?

Reply:
I don't think so, at least this way. The only way to do it "real" is to create a QUnit test that simulates your navigation in Odoo web client.